pkg/clients/metadata/mocks/DeviceClient.go

```diff
@@ -0,0 +1,432 @@
+// Code generated by mockery v1.0.0
```

Comment:
We discussed this a couple weeks ago on the Core WG call. We decided for now to check it in because it would allow the DeviceServices to also utilize the mock for unit tests there. If it's a pain, we can re-evaluate.